clients/rospy/src/rospy/core.py

```diff
@@ -149,6 +151,57 @@ def rospywarn(msg, *args):
 logerror = logerr # alias logerr
 
 logfatal = logging.getLogger('rosout').critical
+
+
+class LoggingThrottle(object):
+
+    last_logging_time_table = {}
+
+    def __call__(self, id, logging_func, period, msg):
+        """Do logging specified message periodically.
+
+        - id (str): Id to identify the caller
+        - logging_func (function): Function to do logging.
+        - period (float): Period to do logging in second unit.
+        - msg (object): Message to do logging.
+        """
+        now = rospy.Time.now()
+
+        last_logging_time = self.last_logging_time_table.get(id)
```

Comment:
But currently I don't know how to get the called location with information about lines and columns.
`inspect.stack()[1][1:]` only returns the line and line number.